src/daemon/signal-handler.c

```diff
@@ -264,7 +264,9 @@ void nd_process_signals(void) {
             last_update_mt += save_every_ut;
         }
 
-        poll(NULL, 0, 13 * MSEC_PER_SEC + 379);
+        if(poll(NULL, 0, 13 * MSEC_PER_SEC + 379) < 0)
+            sleep(13);
```

Comment:
This delays the shutdown of an agent started from the command line and doing a control-c 13 seconds

Reply:
thanks